src/main/java/com/internet/shop/security/AuthenticationImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.internet.shop.security;
+
+import com.internet.shop.exceptions.AuthenticationException;
+import com.internet.shop.lib.Inject;
+import com.internet.shop.lib.Service;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDb = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect login or password."));
+        if (userFromDb.getPassword().equals(password)) {
+            return userFromDb;
+        }
+        throw new AuthenticationException("Incorrect login or password.");
```

Comment:
either make both checks in one if() or make exception message a constant

Reply:
done